kube/client.go

```diff
@@ -148,41 +160,183 @@ func isCompatible(clientMajor, clientMinor, serverMajor, serverMinor string) err
 	return nil
 }
 
-// Apply attempts to "kubectl apply" the file located at path.
-// It returns the full apply command and its output.
-func (c *Client) Apply(path, namespace string, dryRun bool) (string, string, error) {
-	args := []string{"kubectl", "apply", "--validate=false", fmt.Sprintf("--dry-run=%t", dryRun), "-R", "-f", path, "--prune", fmt.Sprintf("-l %s!=%s", c.Label, Off), "-n", namespace}
+func prepareApplyArgs(path, namespace, label string, dryRun bool) []string {
+	args := []string{"kubectl", "apply", fmt.Sprintf("--dry-run=%t", dryRun), "-R", "-f", path, "--prune", fmt.Sprintf("-l %s!=%s", label, Off), "-n", namespace}
 	for _, w := range pruneWhitelist {
 		args = append(args, "--prune-whitelist="+w)
 	}
-	if c.Server != "" {
-		args = append(args, fmt.Sprintf("--kubeconfig=%s", kubeconfigFilePath))
-	}
+
+	return args
+}
+
+func executeApply(args []string) (string, string, error) {
 	cmd := strings.Join(args, " ")
-	stdout, err := exec.Command(args[0], args[1:]...).CombinedOutput()
+	stdout, err := execCommand(args[0], args[1:]...).CombinedOutput()
 	if err != nil {
 		err = fmt.Errorf("Error: %v", err)
 	}
 	return cmd, string(stdout), err
 }
 
+// Apply attempts to "kubectl apply" the file located at path.
+// It returns the full apply command and its output.
+func (c *Client) Apply(path, namespace string, dryRun bool) (string, string, error) {
+	args := prepareApplyArgs(path, namespace, c.Label, dryRun)
+	if c.Server != "" {
+		args = append(args, fmt.Sprintf("--kubeconfig=%s", kubeconfigFilePath))
+	}
+
+	return executeApply(args)
+}
+
+// StrictApply will attempt to "kubectl apply" the file located at path using a `kube-applier` service account under the given namespace.
+// `kube-applier` service account must exist for the given namespace and must contain a secret that include token and ca.cert.
+// It returns the full apply command and its output.
+func (c *Client) StrictApply(path, namespace string, dryRun bool) (string, string, error) {
+	args := prepareApplyArgs(path, namespace, c.Label, dryRun)
+
+	tempKubeConfigFilepath, tempCertFilepath, err := c.CreateTempConfig(namespace, "kube-applier")
+	if err != nil {
+		return "", "", fmt.Errorf("error creating temp config: %v", err)
```

Comment:
Consider using https://github.com/pkg/errors like `errrors.Wrap(err,"error creating temp config")` 
it has a bit nicer syntax as you don't need format the errors  and gives you stack tracer

Reply:
Hmm, this will arguably mean that I will have to change all the errors in that file (at least) to be consistent and the tests as well, but it might be worthy

Reply:
@ffilippopoulos  Then it's probably not worth the effort :)